osu.Game/Overlays/BeatmapSet/Scores/ScoresContainer.cs

```diff
@@ -44,100 +39,165 @@ public ScoresContainer()
-                {
-                    Alpha = 0,
-                    Margin = new MarginPadding(20)
-                },
+                }
             };
+
+            Loading = true;
         }
 
         [BackgroundDependencyLoader]
         private void load(OsuColour colours)
         {
             background.Colour = colours.Gray2;
-            updateDisplay();
         }
 
-        private bool loading
+        private bool loading;
+
+        public bool Loading
         {
-            set => loadingAnimation.FadeTo(value ? 1 : 0, fade_duration);
+            get => loading;
+            set
+            {
+                loading = value;
+
+                if (value)
+                {
+                    loadingContainer.Show();
+                    contentContainer.Hide();
+                }
+                else
+                {
+                    loadingContainer.Hide();
+
+                    if (scores == null || scores?.Scores.Count < 1)
+                        contentContainer.Hide();
+                    else
+                        contentContainer.Show();
+                }
+            }
         }
 
-        private GetScoresRequest getScoresRequest;
-        private IReadOnlyList<ScoreInfo> scores;
+        private APILegacyScores scores;
 
-        public IReadOnlyList<ScoreInfo> Scores
+        public APILegacyScores Scores
         {
             get => scores;
             set
             {
-                getScoresRequest?.Cancel();
                 scores = value;
 
                 updateDisplay();
             }
         }
 
-        private BeatmapInfo beatmap;
-
-        public BeatmapInfo Beatmap
+        private void updateDisplay()
         {
-            get => beatmap;
-            set
-            {
-                beatmap = value;
+            topScoresContainer.Clear();
 
-                Scores = null;
+            scoreTable.Scores = scores?.Scores.Count > 1 ? scores.Scores : new List<APILegacyScoreInfo>();
+            scoreTable.FadeTo(scores?.Scores.Count > 1 ? 1 : 0);
 
-                if (beatmap?.OnlineBeatmapID.HasValue != true)
-                    return;
+            if (scores?.Scores.Any() == true)
+            {
+                topScoresContainer.Add(new DrawableTopScore(scores.Scores.FirstOrDefault()));
 
-                loading = true;
+                var userScore = scores.UserScore;
 
-                getScoresRequest = new GetScoresRequest(beatmap, beatmap.Ruleset);
-                getScoresRequest.Success += r => Schedule(() => Scores = r.Scores);
-                api.Queue(getScoresRequest);
+                if (userScore != null && userScore.Position != 1)
+                    topScoresContainer.Add(new DrawableTopScore(userScore.Score, userScore.Position));
             }
+
+            Loading = false;
         }
 
-        private void updateDisplay()
+        private class ContentContainer : VisibilityContainer
```

Comment:
i don't like the existence of this class. specifically all the custom maxHeight logic, and the fact that the ResizeHeightTo in UpdateAfterChildren is going to fight with the `PopIn`/`PopOut` (I can see the early return, but still ugly). We should not be adding local implementations like this all over the place.

I propose you remove all this custom logic and get a base implementation working, then we can discuss how to make the transitions appear better.